prompting/utils/async_wiki_utils.py

```diff
@@ -0,0 +1,159 @@
+import aiohttp
+import asyncio
+import random
+import bittensor as bt
+from dataclasses import dataclass, field
+from typing import List, Dict
+from tqdm.asyncio import tqdm
+
+EXCLUDE_HEADERS = ("See also", "References", "Further reading", "External links")
+EXCLUDE_CATEGORIES = ("articles", "wiki", "pages", "cs1")
+
+
+class SectionNotFoundException(Exception):
+    """Exception raised when no valid section is found."""
+
+    pass
+
+
+class MaxRetriesReachedException(Exception):
+    """Exception raised when maximum retry attempts are reached."""
+
+    pass
+
+
+@dataclass
+class Context:
```

Comment:
Is this a duplicate of the main context class?

Reply:
It actually is! good catch, will clean up the redundancy